src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -1,15 +1,318 @@
-import { Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { Popover, ColorPicker } from '@wordpress/components';
+
+import { store as nfdOnboardingStore } from '../../../store';
+import { getGlobalStyles, getThemeColors } from '../../../utils/api/themes';
+import { useGlobalStylesOutput } from '../../../utils/global-styles/use-global-styles-output';
+
 const DesignColors = () => {
-	return (
-		<div style={{ padding: '0 16px' }}>
-			<h2>{__('Pick Color Palette', 'wp-module-onboarding')}</h2>
-			<p>
-				{__(
-					'If user has opted for custom design, panel will show color palettes to affect preview pane.',
-					'wp-module-onboarding'
+
+	const [isLoaded, setIsLoaded] = useState(false);
+	const [globalStyles, setGlobalStyles] = useState();
+	const [selectedColors, setSelectedColors] = useState();
+	const [showColorPicker, setShowColorPicker] = useState(false);
+	const [isAccordionClosed, setIsAccordionClosed] = useState(true);
+	const [selectedColorsLocal, setSelectedColorsLocal] = useState();
+
+	const [customColors, setCustomColors] = useState();
+	const [colorPalettes, setColorPalettes] = useState();
+	const [colorPickerCalledBy, setColorPickerCalledBy] = useState('');
+
+	const { storedPreviewSettings, currentData } = useSelect(
+		(select) => {
+			return {
+				storedPreviewSettings:
+					select(nfdOnboardingStore).getPreviewSettings(),
+				currentData:
+					select(nfdOnboardingStore).getCurrentOnboardingData(),
+			};
+		},
+		[]
+	);
+
+	const { updatePreviewSettings, setCurrentOnboardingData } =
+		useDispatch(nfdOnboardingStore);
+
+	function stateToLocal(selectedColors) {
+		if (selectedColors) {
+			let selectedColorsLocalTemp = {};
+			selectedColors?.color.forEach(color => {
+				selectedColorsLocalTemp[color.slug] = color.color;
+			});
+
+			setSelectedColorsLocal(selectedColorsLocalTemp);
+			return selectedColorsLocalTemp;
+		}
+	}
+
+	function LocalToState(selectedColorsLocalTemp, colorStyle) {
```

Comment:
```suggestion
	function localToState(selectedColorsLocal, colorStyle) {
```
Let's try removing `temp` as function argument names in any other places as well